### DISCUSSION TITLE
Use `:undojoin` for inline editing.

### DISCUSSION THREAD

**Discussion by GordianDziwis** [olimorris/codecompanion.nvim#325] [Ideas]
Right now, the answer of the LLM is streamed into the buffer, without joining all the edits with `:undojoin`. So to undo the edit `u` has to be pressed many times.

**Comment by olimorris** [comment]
This was previously in the plugin but it caused a few issues and was not as seamless as I hoped. You can do `gr` to reject the edit ([ref](https://github.com/olimorris/codecompanion.nvim?tab=readme-ov-file#pencil2-inline-assistant))

**Reply by GordianDziwis** [reply]
With `gr` the "streaming" edits are still in the undo history. Can you point me to the function in codecompanion, where the inserts are happening?

**Comment by GordianDziwis** [comment]
Currently, a workaround is to insert the edits with the `@editor` tool from a separate window.

**Reply by olimorris** [reply]
I can't recall when but I had undojoin implemented in the inline strategy file
